scopesim/tests/tests_effects/test_SkycalcTERCurve.py

```diff
@@ -53,3 +54,72 @@ def test_initialises_with_non_skycalc_keys(self):
     def test_initialise_with_local_skycalc_file(self):
         sky_ter = SkycalcTERCurve(use_local_skycalc_file="skycalc_override.fits")
         assert sky_ter.skycalc_table is not None
+
+
+class TestRescalingAndUnitConversion:
+    """Test rescaling of SkyCalcTERCurves and unit conversion."""
+
+    def test_convert_to_nm(self):
+        """Can we idempotent convert to nm?"""
+        tc = SkycalcTERCurve(
+            observatory="armazones",
+            wmin=0.7,
+            wmax=2.5,
+            wunit="um",
+        )
+        tc.convert_to_nm()
+        wmin1 = tc.meta["wmin"]
+        wmax1 = tc.meta["wmax"]
+        wunit1 = tc.meta["wunit"]
+        assert wmin1 == 700
+        assert wmax1 == 2500
+        assert wunit1 == "nm"
+        tc.convert_to_nm()
+        assert wmin1 == 700
+        assert wmax1 == 2500
+        assert wunit1 == "nm"
+
+    def test_save_to(self):
+        """Test whether exact unit conversion works."""
+        for ufrom, uto, fac in [
+            (u.um, u.nm, 1000.),
+            (u.mm, u.um, 1000.),
+            (u.nm, u.um, 0.001),
+            (u.um, u.mm, 0.001),
+        ]:
+            assert save_unit_to(ufrom, uto) == fac
+
+    def test_rescaling(self):
+        """Test rescaling of SkyCalcTERCurves and unit conversion.
+
+        There is a unit conversion problem that prevents rescaling of
+        SkyCalcTERCurves.
+
+        When wunit="um" and wmax=2.5, then this gets converted to nm by astropy,
+        because that is what Skycalc expects. However,
+        u.um.to(u.nm) = 999.9999999999999
+        and therefore 2.5 um is converted to 2499.99 nm.
+
+        2499.99 nm is less than 2.5 um, so there is no overlap anymore between
+        the source spectrum and the to-be-scaled to spectrum, leading to an
+        error.
```

Comment:
Does it *still* "lead to an error"? Isn't that what was fixed here? Same for the comment at the bottom of this method.